### ISSUE TITLE
How to undo %ignore ""

### ISSUE THREAD

**Issue by cminyard** [swig/swig#2173]
I want to ignore everything in a particular file except for one type, but then I want to process some more files.  So I have:
```
%ignore "";
%rename("%s") gensio_time;
%include <gensio/gensio_types.h>
%rename("") "";
```
but the last %rename there doesn't undo the %ignore like I would expect.  Everything is still ignored.  If you can tell me how to do this, I'll modify the docs and send you a pull request.  Thanks.

**Comment by ojwb** [comment]
Hmm, I'd have expected that to work.  Does this?

```
%rename("%s") "";
```

If not, I'll probably have to poke around a bit.

**Comment by cminyard** [comment]
Yes,
```
%rename("%s") "";
```
did work.  Should I submit a doc update?

**Comment by ojwb** [comment]
Yes, it would be helpful to explicitly say you can completely override an earlier `%rename` or `%ignore` by renaming the same target to `"%s"`.  The use of `"%s"` is covered, but only to unignore specific cases of a broader `%ignore` it seems.

**Comment by ojwb** [comment]
BTW if you're enjoying writing docs about `%rename`, there's also #193 (see last comment).

**Comment by cminyard** [comment]
I actually don't mind doing docs.  I prefer coding, of course, but docs make life better in the long run.  Considering how complex SWIG is, the docs are pretty good, and it's really hard to write good docs.   So I have a fix queued up for #193.
